### ISSUE TITLE
Design choice of trainable weights

### ISSUE THREAD

**Issue by summelon** [MedicineToken/Medical-SAM2#14]
Hi guys,
Thank you for your excellent work.

I am a bit confused about the selection of which parts of weights should be fine-tuned.
To clarify, I listed the trainable weights I observed for comparison:
- In your previous work, Medical-SAM-Adapter
  - The trainable weights are adapter layers in the encoder and full mask decoder
- In this work
  - For 2D training: the entire model is trainable
  - For 2D training: only the mask decoder and the memory-relative components are trainable

Could you please explain the rationale behind these decisions?

**Comment by jiayuanz3** [comment]
In this work, we train subparts except prompt encoder. Because we mention the terminology `**one-prompt**' proposed in the paper, so the occurrence of prompts should be sparse. That's why we froze prompt encoder. The image embedding's quality has an influence on memory quality, that's why we train both. For MSA, it attempts to efficiently adapt to the medical setting, so only train part of the layers in the image encoder.  As medical image varies a lot to natural image (e.g. modality, ambiguous boundary etc.), we train the mask decoder in both MSA and MedSAM-2. In short, the scope and purpose of these two methods are different, so we make different decisions at training components. Also, you can freely freeze or defreeze any part depends on your need. Hope that helps!

**Comment by summelon** [comment]
> In this work, we train subparts except prompt encoder. Because we mention the terminology `**one-prompt**' proposed in the paper, so the occurrence of prompts should be sparse. That's why we froze prompt encoder. The image embedding's quality has an influence on memory quality, that's why we train both. For MSA, it attempts to efficiently adapt to the medical setting, so only train part of the layers in the image encoder. As medical image varies a lot to natural image (e.g. modality, ambiguous boundary etc.), we train the mask decoder in both MSA and MedSAM-2. In short, the scope and purpose of these two methods are different, so we make different decisions at training components. Also, you can freely freeze or defreeze any part depends on your need. Hope that helps!

Thank you for your reply
I apologize for the mistake in my question (updated).
I am trying to figure out: in this work, why the image encoder is trainable in 2D training but frozen in 3D training?

**Comment by jiayuanz3** [comment]
That's depends on your need. We find training 3D image encoder requires much more computational resource compared to 2D. Of course, fine-tune the entirely whole model is more likely for better performance (task specific). But considering the resource, we made such training decision.

**Comment by summelon** [comment]
Very clear.
Thank you so much.
